lib/functions_framework/server.rb

```diff
@@ -306,7 +306,7 @@ def min_threads
       # @return [Integer]
       #
       def max_threads
-        @max_threads || 8
+        @max_threads || 1024
```

Comment:
Do you think it's necessary to also add some buffer to @max_threads here?

Reply:
Probably, Puma takes some threads by default: https://github.com/puma/puma#thread-pool

Reply:
I did run the conformance concurrency test with 1000 requests and it passed with 1024 threads so decided to go ahead with this number.

_"Be aware that additionally Puma creates threads on its own for internal purposes (e.g. handling slow clients). So, even if you specify -t 1:1, expect around 7 threads created in your application."_ - Looks like Puma adds extra threads beyond the specified limit if needed.

Reply:
Tested this code path by setting the --max-thread=x parameter and setting the x to 10, 20, 50, 100, 200.. and the tests worked without needing any buffer.